internal/rest/avi_obj_vsvip.go

```diff
@@ -119,9 +119,16 @@ func (rest *RestOperations) AviVsVipBuild(vsvip_meta *nodes.AviVSVIPNode, vsCach
 						if vsvip.Vip[0].IPAMNetworkSubnet == nil {
 							vip.IPAMNetworkSubnet = &avimodels.IPNetworkSubnet{}
 						} else {
-							vip.IPAMNetworkSubnet = vsvip.Vip[0].IPAMNetworkSubnet
+							var ipamNetworkRefSplit []string
+							if vsvip.Vip[0].IPAMNetworkSubnet.NetworkRef != nil {
+								ipamNetworkRefSplit = strings.Split(*vsvip.Vip[0].IPAMNetworkSubnet.NetworkRef, "#")
+							}
+							if len(ipamNetworkRefSplit) > 1 && ipamNetworkRefSplit[1] != vsvip_meta.VipNetworks[0].NetworkName {
+								vip.IPAMNetworkSubnet = &avimodels.IPNetworkSubnet{}
```

Comment:
can we refactor these conditions from line 119 to line 134 for better readability?
the condition is reduced to "use old subnet only if networkName matches"